liferay-objects/liferay-portal
# LPS-176408 Adds one to many and many to many info collection providers for objects

## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPS-176408)

Dear Team, this PR adds 2 new related info collection providers for object relationship. One for One to Many, the other for Many to Many. Please see the parent LPS for more details on it. I have also included two gifs for the feature. One is for many to many, the other for one to many relationship

## Change summary:

* Adds 2 new related info collection providers for object relationship
* Register the providers in object deployer

## Notes

Feature is behind epic, to enable: feature.flag.LPS-176083=true

## Test Scenario One to Many

* Create 2 objects and establish a one to many relationship between them
* Add a new collection provider to the page and select the collection provider for the parent object definition
* Add a collection provider inside the previous collection provider, and configure it to show the relationship
* When the page is published the information is shown correctly

![onetomany](https://user-images.githubusercontent.com/4267448/220665484-57c3b948-2985-42bc-a774-e032e86ce754.gif)


## Test Scenario Many to Many

* Create 2 objects and establish a many to many relationship between them
* Add a new collection provider to the page and select the collection provider for the parent object definition
* Add a collection provider inside the previous collection provider, and configure it to show the relationship
* When the page is published the information is shown correctly

![manytomany](https://user-images.githubusercontent.com/4267448/220662968-5d6ac7c9-c6a8-4782-8e72-16c8bb820f42.gif)


## Notes

* Integration tests and poshi tests are not included and will be sent in a follow up task
